### DISCUSSION TITLE
How can i fix not working of 'Alt+Tab' and 'Window+Tab'?

### DISCUSSION THREAD

**Discussion by surfhawk** [valinet/ExplorerPatcher#1087] [General]
The Alt-Tab function works well, but after executing Window+Tab even once, 
Alt-Tab does not work at all. (And the Window-Tab action doesn't respond at all) Is this a bug?

In this case, the Alt-Tab function returns normally only after deleting Explorer Patcher and reinstalling it.
I'm using Windows 11. Thanks

**Comment by Lieuallen** [comment]
I'm using Windows 11 (22H2 22621.1265) and ExplorerPatcher (22621.819.52.1) and both Alt-Tab and Win-Tab work just fine for me.

**Reply by inphra** [reply]
This applies to Windows 11 (22H2 22621.1413).

**Comment by JasonJasu** [comment]
I'm vouching on this. ExplorerPatcher (Version 22621.1344,53.1) was working fine until I updated windows to the latest version (Version 22H2 OS Build 22621.1413). I can't get the windows shortcut working, like for example Windows+A for Control Center, or Windows+N for notifications. 

I can't Alt+Tab after pressing Windows+Tab, thats something we had similar so i assume It's on ExplorerPatcher. It's a shame i have to disable it in the meantime....

**Reply by 0x01001100** [reply]
Exactly; after using Win+Tab, Alt+Tab will stop working..... till you restart the explorer

**Comment by AlexAdasCca** [comment]
In version 22621.1413, Microsoft officially announced the results of the last A/B experiment on optimizing the taskbar feature code, which is essentially aimed at fixing the UI display abnormality caused by the automatic hiding of the taskbar function on tablets. Due to the complete changes in the part of the taskbar UI in the UWP framework adopted by Explorer, all taskbar transparency beautification software, including Explorer Pacher, is now invalid. "I believe that in the future, our software developers will be able to synchronously fix this issue. This is not a problem in itself, it's just that Microsoft has changed the code."



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月17日(星期五) 中午1:53
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; 
主题: Re: [valinet/ExplorerPatcher] How can i fix not working of &#39;Alt+Tab&#39; and &#39;Window+Tab&#39;? (Discussion #1087)





 
Exactly; after using Win+Tab, Alt+Tab will stop working..... till you restart the explorer

**Reply by HW71** [reply]
Hi all,

hmm, for me also the desktop switching functionality is not working any more... So the only solution seems to be to uninstall Explorer Patcher, right?

Thanks so mich inh advance & kind regards!

-Erik

**Reply by AlexAdasCca** [reply]
It seems that you can use the ViveTool app to disable the feature change with id 26008830, which allows you to continue using Explorer Patcher, which is a mitigation solution for the transition period. I think this is feasible.

**Reply by AlexAdasCca** [reply]
You can find it in Github, which is open source and the GUI-project author is PeterStrike, and the original author is bookishclosed.

**Reply by AlexAdasCca** [reply]
https://github.com/thebookisclosed/ViVe

**Reply by AlexAdasCca** [reply]
https://github.com/PeterStrick/ViVeTool-GUI

**Comment by AlexAdasCca** [comment]
The KB5023706 version update will synchronize the installation of the KB5022913 update in the preview version during the third phase of DCOM security (mandatory phase). You can check the details of the KB5022913 update, which will provide all the answers.

**Comment by AlexAdasCca** [comment]
Let's take a look at similar issues in Translucent TB software. Our solution is as follows:
The new update brings a brand new taskbar called Foldable Taskbar. It have a pointer to judge the taskbar is folded. But old version dont have this function, so it's a null pointer. There is a dilemma in choosing between brand new taskbar without any bug and Translucent TB with a crush while right click on Windows. To use new taskbar, enter "vivetool addconfig 26008830 2" in vivetool. Enter "vivetool delconfig 26008830 2" in vivetool to use old taskbar.

**Comment by HW71** [comment]
Hi all,

I just used 

https://github.com/PeterStrick/ViVeTool-GUI

as suggested by @CthRio and disabled feature 26008830. Afterwards I had to restart my machine and now my Taskbar is back working as normal. That is, I can use Alt+Tab to switch between applications and also the "Active Applications" button in the taskbar works again, so I can move application between different virtual desktops!

Thanks again, @CthRio, for the hint!

Kind regards, Erik

**Reply by AlexAdasCca** [reply]
I'm glad I could help.But thanks to the author of the Vive project.

**Comment by ishalev-me** [comment]
Same problem, also used ViVeTool to fix for now.

**Comment by Leokuma** [comment]
I downloaded the [ViVeTool CLI](https://github.com/thebookisclosed/ViVe/releases/download/v0.3.2/ViVeTool-v0.3.2.zip), ran the command below as Admin, _restarted my PC_ and now everything is working again (<kbd>Alt+Tab</kbd>, <kbd>Win+Tab</kbd>, volume indicator etc):

```
.\ViVeTool.exe /disable /id:26008830
```

Thank you @CthRio.

**Reply by AlexAdasCca** [reply]
This is only a temporary solution. Future system version updates may also result in the invalidation of relevant functions on the taskbar, and then it is necessary to use this tool or other methods again. Moreover, I think it is necessary to disassemble the sequence and find the cause of the problem, otherwise it will be troublesome to disable some features after every important Windows update.

**Reply by valinet** [reply]
The feature is apparently called `STTest` (it is named so in vivetool GUI). Searching through `explorer`'s symbols, there are mentions of this, with a couple of functions checking if this feature is available etc. The problem, as usual, is how to hook this function and change its return... the app needs rearchitecting badly, modularize the design, ease the use of symbols... a lot of work actually, and knowing this makes investing in the existing code base hard.

**Reply by asaf400** [reply]
Can confirm this works after restart, Good job everyone..

Edit: removed unnecessary note about volume indicator

**Reply by KyleKolander** [reply]
Fixed it for me, too!  Thanks!

**Reply by kornylak** [reply]
Thanks! I have searched for months trying to fix this!!

**Comment by Anixx** [comment]
Likely, STTest stands for "Surface Taskbar Test".

**Comment by AlexAdasCca** [comment]
Microsoft actually gave a vague explanation, see [here](https://learn.microsoft.com/en-us/windows/release-health/status-windows-11-22h2#third-party-ui-customization-apps-might-cause-windows-to-not-start-up)[Third-party UI customization apps might cause Windows to not start up].

**Comment by piradix** [comment]
Hello, when did you think do un update  whothout desactivate "taskbar optimisation" ?

**Reply by AlexAdasCca** [reply]
This occurred after the recent public release of OS Build 22621.1344 (KB5023706), along with the installation of KB5022913. However, this may have occurred in earlier tests of the Preview version(KB5022913), and both third-party software developers and Microsoft engineers are investigating this issue.

**Reply by piradix** [reply]
Can't you avoid this MS bug? Because MS seems to say it's an unreferenced api (If I understand correctly they write).

Nota: you had do a good job ;-)

**Reply by topia** [reply]
ExplorerPatcher uses undocumented APIs and patches binary code by design. You should accept it when you use EP.

**Comment by asaf400** [comment]
Fixed in latest version as of this message:
https://github.com/valinet/ExplorerPatcher/releases/tag/22621.1413.54.1_8b5443d

**Reply by jwgn** [reply]
Not fixed. I just updated to that version and Alt-Tab / Win-Tab is still not working properly.

22H2 22623.1391

**Reply by topia** [reply]
ExplorerPatcher doesn't support insider builds.

**Reply by jwgn** [reply]
Thanks, did not know I was on a beta branch. After inplace upgrade(downgrade?) to 22621.1413 everything works fine with the latest update.

**Comment by piradix** [comment]
For me, with the new version (Version 22621.1413.543), all work : ALT6TAB or MS+TAB
Good job
thank's

**Comment by surfhawk** [comment]
In recent update (Version 22621.1413.543),
This problem is resolved. Thanks!

**Comment by Hrhigher** [comment]
Hello with compil 22621.2070 the issue arose again! I tried already using ViveTool but no luck! Is there any other workaround?

**Reply by pyrates999** [reply]
That is a preview update: https://support.microsoft.com/en-us/topic/july-26-2023-kb5028254-os-build-22621-2070-preview-66161ea6-c106-43e9-bef5-8410ccbd34e8

Uninstall it if you want to use explorer patcher and don't install preview updates in the future: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

**Comment by Azarien** [comment]
Unfortunately the issue occurs on 2261.2134 (which is documented to include the "improvements" of KB5028245, see https://support.microsoft.com/en-us/topic/august-8-2023-kb5029263-os-build-22621-2134-f8d4d3de-47c1-40e1-a2e6-97c2770ee2e8)

Disabling feature 26008830 with ViVe does not help (I guess it can no longer be disabled).
Note that the problem somehow depends on Windows 10 taskbar. When using Windows 11 taskbar it works fine. But classic taskbar is the main reason that I want to use ExplorerPatcher.

**Comment by asaf400** [comment]
@Azarien 
it works for me (22621.2134), but EP did crash first time around, 
I had to do this to fix it, maybe you should try that as well and see if it works:
https://github.com/valinet/ExplorerPatcher/discussions/1702#discussioncomment-6580555

edit: don't worry about the keys, it's not the EP settings \ options, those are saved elsewhere, so EP registry is safe to nuke

**Reply by asaf400** [reply]
@Azarien you know what? You may be right, Alt+Tab works perfectly,
until a Win+Tab is pressed, that doesn't work and then they are both broken..

edit: a restart for EP fixes Alt+Tab

**Comment by Usv3re** [comment]
Bumping this, having same issue. Latest update didn't fix this
